apps/web/src/components/Feed/FeedList.tsx

```diff
@@ -56,100 +46,20 @@ export default function FeedList({
-        return feedDataRef.current[feedDataRef.current.length - rowIndex - 1]?.dbid;
-      },
-      defaultHeight: 400,
-      fixedWidth: true,
-      minHeight: 0,
-    });
-  }, []);
-
-  // Function responsible for tracking the loaded state of each row.
-  const isRowLoaded = useCallback(
-    ({ index }: { index: number }) => !hasNext || !!feedData[index],
-    [feedData, hasNext]
-  );
-
-  const virtualizedListRef = useRef<List | null>(null);
-
-  const handlePotentialLayoutShift = useCallback(
-    (index: number) => {
-      measurerCache.clear(index, 0);
-      virtualizedListRef.current?.recomputeRowHeights(index);
-    },
-    [measurerCache]
-  );
+  const parentRef = useRef<HTMLDivElement>(null);
+  const parentOffsetRef = useRef(0);
 
-  //Render a list item or a loading indicator.
-  const rowRenderer = useCallback(
-    ({
-      index,
-      key,
-      parent,
-      style,
-    }: {
-      index: number;
-      key: string;
-      style: React.CSSProperties;
-      parent: MeasuredCellParent;
-    }) => {
-      if (!isRowLoaded({ index })) {
-        return <div />;
-      }
-      // graphql returns the oldest event at the top of the list, so display in opposite order
-      const content = feedData[feedData.length - index - 1];
+  const virtualizer = useWindowVirtualizer({
+    count: hasNext ? feedData.length + 1 : feedData.length,
+    estimateSize: () => 800,
+    scrollMargin: parentOffsetRef.current,
+    overscan: 5,
+  });
 
-      // Better safe than sorry :)
-      if (!content) {
-        return;
-      }
-
-      return (
-        <CellMeasurer
-          cache={measurerCache}
-          columnIndex={0}
-          rowIndex={index}
-          key={key}
-          parent={parent}
-        >
-          {({ registerChild }) => (
-            // @ts-expect-error: this is the suggested usage of registerChild
-            <div ref={registerChild} style={style} key={key}>
-              <FeedEvent
-                // Here, we're listening to our children for anything that might cause
-                // the height of this list item to change height.
-                // Right now, this consists of "admiring", and "commenting"
-                //
-                // Whenever the height changes, we need to ask react-virtualized
-                // to re-evaluate the height of the item to keep the virtualization good.
-                onPotentialLayoutShift={handlePotentialLayoutShift}
-                index={index}
-                eventRef={content}
-                key={content.dbid}
-                queryRef={query}
-                feedMode={feedMode}
-              />
-            </div>
-          )}
-        </CellMeasurer>
-      );
-    },
-    [feedData, feedMode, handlePotentialLayoutShift, isRowLoaded, measurerCache, query]
-  );
+  const feeds = virtualizer.getVirtualItems();
+  useLayoutEffect(() => {
+    parentOffsetRef.current = parentRef.current?.offsetTop ?? 0;
+  }, []);
```

Comment:
Can you explain what this is doing? I tried playing around with it locally to get a feel for it but no matter what I change `scrollMargin` to, nothing changes on my screen. This feels ugly and I'm hoping we don't need it 😂

Reply:
+1 would love to have comments in the code for stuff like this!

Reply:
I removed it 😅 . Seems like it doesnt do any changes. lol